invokeai/backend/model_manager/probe.py

```diff
@@ -52,66 +94,51 @@ class ModelProbe(object):
-        if isinstance(model, Path):
-            return cls.probe(model_path=model, prediction_type_helper=prediction_type_helper)
-        elif isinstance(model, (dict, ModelMixin, ConfigMixin)):
-            return cls.probe(model_path=None, model=model, prediction_type_helper=prediction_type_helper)
-        else:
-            raise InvalidModelException("model parameter {model} is neither a Path, nor a model")
+        :param format: The ModelFormat of the model to be probed.
+        :param model_type: The ModelType of the model to be probed.
+        :param probe_class: The class of the prober (inherits from ProbeBase).
+        """
+        cls.PROBES[format][model_type] = probe_class
 
     @classmethod
     def probe(
         cls,
         model_path: Path,
-        model: Optional[Union[Dict, ModelMixin]] = None,
         prediction_type_helper: Optional[Callable[[Path], SchedulerPredictionType]] = None,
-    ) -> ModelProbeInfo:
-        """
-        Probe the model at model_path and return sufficient information about it
-        to place it somewhere in the models directory hierarchy. If the model is
-        already loaded into memory, you may provide it as model in order to avoid
-        opening it a second time. The prediction_type_helper callable is a function that receives
-        the path to the model and returns the SchedulerPredictionType.
-        """
-        if model_path:
-            format_type = "diffusers" if model_path.is_dir() else "checkpoint"
-        else:
-            format_type = "diffusers" if isinstance(model, (ConfigMixin, ModelMixin)) else "checkpoint"
-        model_info = None
+    ) -> Optional[ModelProbeInfo]:
+        """Probe model."""
         try:
             model_type = (
-                cls.get_model_type_from_folder(model_path, model)
-                if format_type == "diffusers"
-                else cls.get_model_type_from_checkpoint(model_path, model)
+                cls.get_model_type_from_folder(model_path)
+                if model_path.is_dir()
+                else cls.get_model_type_from_checkpoint(model_path)
             )
-            format_type = "onnx" if model_type == ModelType.ONNX else format_type
+            format_type = (
+                "onnx" if model_type == ModelType.ONNX else "diffusers" if model_path.is_dir() else "checkpoint"
+            )
+
             probe_class = cls.PROBES[format_type].get(model_type)
+
             if not probe_class:
                 return None
-            probe = probe_class(model_path, model, prediction_type_helper)
+
+            probe = probe_class(model_path, prediction_type_helper)
+
             base_type = probe.get_base_type()
             variant_type = probe.get_variant_type()
             prediction_type = probe.get_scheduler_prediction_type()
             format = probe.get_format()
+            hash = FastModelHash.hash(model_path)
+
             model_info = ModelProbeInfo(
                 model_type=model_type,
```

Comment:
model_type may be None at this time, but ModelProbeInfo's model_type arg is required